Sources/MapboxMaps/Offline/OfflineCallbacks.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+/// Returns a closure suitable for the OfflineManager and TileStore callback based
+/// APIs, that converts the expected type into a Swift Result type.
+/// - Parameters:
+///   - closure: developer provided completion closure.
+///   - type: The ObjC type. For example, for `[TileRegion]` this would be `NSArray`
+/// - Returns: A suitable `MBXExpected` base closure.
+internal func coreAPIClosureAdapter<T, ErrorType, ObjCType>(for closure: @escaping (Result<T, ErrorType>) -> Void, type: ObjCType.Type)
```

Comment:
I'm thinking that our `Result`s should use weakly-typed errors. I don't like that we're overloading the TileRegionError's "other" case with platform-level errors.

Reply:
Done.